File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/抽象工厂模式/java/factory/ConcreteFactory1.java
package com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.factory;

import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.abst.ProductA;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.abst.ProductB;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.modle.ConcreteProductA1;
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.java.modle.ConcreteProductB1;

public class ConcreteFactory1 extends Factory {

    @Override
    public ProductA createProductA() {
        return new ConcreteProductA1();
    }

    @Override
    public ProductB createProductB() {
        return new ConcreteProductB1();
    }
}

File: app/src/main/java/com/redkey/kotlintest/kotlin/design_pattern/抽象工厂模式/kotlin/factory/ConcreteFactory1.kt
package com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.factory

import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.abst.ProductA
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.abst.ProductB
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.modle.ConcreteProductA1
import com.redkey.kotlintest.kotlin.design_pattern.抽象工厂模式.kotlin.modle.ConcreteProductB1

class ConcreteFactory1 : Factory() {
    override fun createProductA(): ProductA{
        return ConcreteProductA1()
    }

    override fun createProductB(): ProductB{
        return ConcreteProductB1()
    }
}